incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/openapi/OpenApiHttpProxyConfigGenerator.java

```diff
@@ -0,0 +1,366 @@
+import static org.agrona.LangUtil.rethrowUnchecked;
+
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import jakarta.json.Json;
+import jakarta.json.JsonArrayBuilder;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonPatch;
+import jakarta.json.bind.Jsonb;
+import jakarta.json.bind.JsonbBuilder;
+import jakarta.json.spi.JsonProvider;
+
+import io.aklivity.zilla.runtime.binding.http.config.HttpConditionConfig;
+import io.aklivity.zilla.runtime.binding.http.config.HttpOptionsConfig;
+import io.aklivity.zilla.runtime.binding.tcp.config.TcpOptionsConfig;
+import io.aklivity.zilla.runtime.binding.tls.config.TlsOptionsConfig;
+import io.aklivity.zilla.runtime.command.config.internal.airline.ConfigGenerator;
+import io.aklivity.zilla.runtime.command.config.internal.openapi.model.OpenApi;
+import io.aklivity.zilla.runtime.command.config.internal.openapi.model.PathItem;
+import io.aklivity.zilla.runtime.command.config.internal.openapi.model.Server;
+import io.aklivity.zilla.runtime.engine.config.BindingConfigBuilder;
+import io.aklivity.zilla.runtime.engine.config.ConfigWriter;
+import io.aklivity.zilla.runtime.engine.config.GuardedConfig;
+import io.aklivity.zilla.runtime.engine.config.NamespaceConfig;
+import io.aklivity.zilla.runtime.engine.config.NamespaceConfigBuilder;
+import io.aklivity.zilla.runtime.engine.config.RouteConfig;
+import io.aklivity.zilla.runtime.engine.config.RouteConfigBuilder;
+import io.aklivity.zilla.runtime.guard.jwt.config.JwtOptionsConfig;
+import io.aklivity.zilla.runtime.vault.filesystem.config.FileSystemOptionsConfig;
+
+public class OpenApiHttpProxyConfigGenerator implements ConfigGenerator
+{
+    private static final JsonPatch ENV_VARS_PATCH = createEnvVarsPatch();
+
+    private final OpenApi openApi;
+    private final ConfigWriter configWriter;
+
+    public OpenApiHttpProxyConfigGenerator(
+        Path input)
+    {
+        this.openApi = parseOpenApi(input);
+        this.configWriter = new ConfigWriter(null);
+    }
+
+    public String generateConfig()
+    {
+        return writeConfig(createNamespaceConfig());
+    }
+
+    private OpenApi parseOpenApi(
+        Path input)
+    {
+        OpenApi openApi = null;
+        try (InputStream inputStream = new FileInputStream(input.toFile()))
+        {
+            Jsonb jsonb = JsonbBuilder.create();
+            openApi = jsonb.fromJson(inputStream, OpenApi.class);
+            jsonb.close();
+        }
+        catch (Exception ex)
+        {
+            ex.printStackTrace();
+            rethrowUnchecked(ex);
+        }
+        return openApi;
+    }
+
+    private NamespaceConfig createNamespaceConfig()
+    {
+        NamespaceConfigBuilder<NamespaceConfig> namespaceBuilder = NamespaceConfig.builder()
+            .name("example");
```

Comment:
In places where it feels necessary to break the flow to perform some dynamic behavior, it would be more convenient to inject the logic into the flow instead.

For example:
```
    return NamespaceConfig.builder()
        .name("example")
        .inject(this::injectJwtGuard)
        .binding()
            .type("tcp")
...
```

So I think we should add something like `R inject(Function<T, R> injector)` to `ConfigBuilder` subtypes.

Reply:
Yep. I guess the code would look nicer that way. So, the argument here would be a function reference that builds and returns the object to inject, right? What if we need to insert more than one element, could we handle that case, too? 🤔